modules/message-lane/src/inbound_lane.rs

```diff
@@ -55,29 +55,32 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 	/// Receive state of the corresponding outbound lane.
 	pub fn receive_state_update(&mut self, outbound_lane_data: OutboundLaneData) -> Option<MessageNonce> {
 		let mut data = self.storage.data();
-		if outbound_lane_data.latest_received_nonce > data.latest_received_nonce {
+		let last_received_nonce = data.last_delivered_nonce();
+
+		if outbound_lane_data.latest_received_nonce > last_received_nonce {
 			// this is something that should never happen if proofs are correct
 			return None;
 		}
-		if outbound_lane_data.latest_received_nonce <= data.latest_confirmed_nonce {
+		if outbound_lane_data.latest_received_nonce <= data.last_confirmed_nonce {
```

Comment:
Same here

Reply:
That's outbound lane data, I decided to only change `InboundLaneData` to be able to distinguish variables without knowing the inbound/outbound context.